### PR TITLE
Use Python 3 syntax in utilities

### PR DESCRIPTION
**Pull request by MartinThoma**
#### References to other Issues or PRs

* https://github.com/sympy/sympy/issues/18816
* https://github.com/sympy/sympy/pull/19194

#### Brief description of what is fixed or changed

Changed Python 2 syntax to Python 3 syntax:

* IOError -> OSError: https://stackoverflow.com/a/29347946/562769
* Remove __future__ imports
* Remove explicit inheritance from object
* super(SomeClass, self) -> super()
* String Formatting
    * Percentage to format
    * {0} -> {}
* set([]) -> set()
* Use `yield from`

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Sc0rpi0n101** [comment]
You have unused imports in `test_pickling.py`

**Comment by oscarbenjamin** [comment]
Are you using pyupgrade for this? There were other PRs using that and we agreed to leave `%`-style formatting. There is a command line option to pyupgrade to do that.

**Comment by smitgajjar** [comment]
As suggested by @oscarbenjamin, you can use this command:
`find -name "*.py" | xargs pyupgrade --py3-plus --keep-percent-format`

File: sympy/utilities/autowrap.py [hunk 1]
```diff
@@ -65,7 +65,6 @@
 
 """
 
-from __future__ import print_function, division
```
Comment (smitgajjar) [inline sympy/utilities/autowrap.py:68]: Remove this newline from here as well as elsewhere. Keep only one newline between docstring and imports.

File: sympy/utilities/codegen.py [hunk 2]
```diff
@@ -79,7 +79,6 @@
 
 """
```
Comment (smitgajjar) [inline sympy/utilities/codegen.py:81]: Here too!

**Comment by MartinThoma** [comment]
> Are you using pyupgrade for this? There were other PRs using that and we agreed to leave %-style formatting. There is a command line option to pyupgrade to do that.

Ok, I didn't know that.

Yes, I do use pyupgrade (+ some manual checks / corrections in cases where I thought the percentage notation was clear).

I'll revert the percentage changes... might take a bit.

File: sympy/utilities/autowrap.py (hunk 1 above)
Comment (MartinThoma) [reply sympy/utilities/autowrap.py:68]: An automatic code formatter like `black` would be awesome for things like this

**Comment by Sc0rpi0n101** [comment]
Is anyone keeping track of which modules/files have been covered and which ones are left?
I hope this is not being done randomly.

**Comment by MartinThoma** [comment]
> I hope this is not being done randomly.

I wanted to do this by sub-packages. One could run it over the complete codebase, but this would be a nightmare to review (and also quite a bit of work for the initial PR as I manually check the changes). I had one PR for the vector subpackage (see link above), utils was next.

**Comment by oscarbenjamin** [comment]
Take a look at #18877 and others (also look at the closed labels):
https://github.com/sympy/sympy/labels/Dropping%20Python%202

You might also be interested in these:
https://github.com/sympy/sympy/labels/Code%20quality

**Comment by smitgajjar** [comment]
Pyuprade is carried out until `ntheory` (PR #18825 #18877 #19160) and @MartinThoma covered `vector`. Please note **Other Comments** section in these PRs

File: sympy/utilities/autowrap.py (hunk 1 above)
Comment (MartinThoma) [reply sympy/utilities/autowrap.py]: Fixed all of those manually :-)

**Comment by MartinThoma** [comment]
I also searched for those:

```
grep -rnIi "python2" *
grep -rnIi "python 2" *
grep -rnIi "py2" *
```

**Comment by oscarbenjamin** [comment]
Looks like the tests failed because of a network problem. I've restarted them

**Comment by oscarbenjamin** [comment]
Looks good. Thanks!

**Pull request merged**